### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements to sd-flag

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1611]
## Description
This task describes the Storybook implementation for the sd-flag component as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for sd-flag can be found here: [sd-flag Notes](https://www.notion.so/matuzo/sd-flag-111f3f6e1bc080eab710cc694764013c).

-----------------

## Comment
**Problem:** The contrast ratio in sd-flag--neutral-500 is not high enough (2.21:1).
**Solution:** Use a ratio of at least 4.5:1.

sd-flag V.1.0.1
Fix: Change text color on bg neutral-500 from white to black to comply with WCAG 2.2
See [figma hand-off](https://www.figma.com/design/YDktJcseQIIQbsuCpoKS4V/Component-Docs?node-id=2106-34540&node-type=frame&t=LMmzVkP8mWhXfDVs-11)

## Note
- The heading in a teaser should be the first item in the DOM. A heading introduces a new thematic region and separates the following content from the previous region. Although chips and flags are visually the first items in the teaser, they should come after the heading in the DOM.

## Related WCAG criteria
- https://www.w3.org/WAI/WCAG22/quickref/#meaningful-sequence

## Template Ticket
- [ ] Pick up the template ticket when working on the component story for A11y improvement, as it references the component and will be resolved once the component is made accessible:
#1530 

## DoR
- [x] Item has business value
- [x] Item has been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

**Comment by coraliefeil** [comment]
I’ve updated sd-flag and the issue description.

sd-flag V.1.0.1
Fix: Change text color on bg neutral-500 from white to black to comply with WCAG 2.2
See [figma hand-off](https://www.figma.com/design/YDktJcseQIIQbsuCpoKS4V/Component-Docs?node-id=2106-34540&node-type=frame&t=LMmzVkP8mWhXfDVs-11)
